File: localized/ja/01-Navigation/1-Navigate_to_something_you_know/1.5-Go_to_file_member.js
// ファイルメンバーへの移動 (JavaScript)
//
// 現在のファイルで定義されているメンバーに移動します。
//
// <shortcut id="File Member">Alt+\    (ReSharper VisualStudio Keymap)</shortcut>
// <shortcut id="File Member">Ctrl+F12 (Rider Default IntelliJ Keymap)</shortcut>
//
// 1. "example" 、または "other" とタイプして移動してみましょう。
//
// 2. 中間一致とワイルドカードを使って移動してみましょう。
//    以下の様にタイプして動きを確認してみましょう。
//      "interesting"
//      "really*function"
//
// 3. アンダースコアで区切られた名前に移動する方法を確認してみます。
//    "rif" とタイプしてみましょう。
//    古いバージョンの場合大文字小文字を区別します - "RIF" ではありません。
//    (参考) Rider 2021.1.2 では "RIF" でもマッチします。

function example(name, age) {
    // ...
}

function other() {
    // ...
}

function really_interesting_function() {
    // ...
}

File: localized/ja/01-Navigation/1-Navigate_to_something_you_know/1.5-Go_to_file_member.cs
namespace JetBrains.ReSharper.Koans.Navigation
{
    // ファイルメンバーへの移動
    //
    // 現在のファイルで定義されているメンバーに移動します。
    //
    // <shortcut id="File Member">Alt+\    (ReSharper VisualStudio Keymap)</shortcut>
    // <shortcut id="File Member">Ctrl+F12 (Rider Default IntelliJ Keymap)</shortcut>
    //
    // 1. 名前、CamelHumps (略語入力)、中間一致やワイルドカードで移動します。
    //    以下のようなキーワードをタイプして結果を確認してみましょう。
    //      "GoToFileMemberExample"
    //      "GTFME"
    //      "FileMember"
    //      "GoTo*Example"
    //
    // 2. 名前で Class メンバーに移動します。
    //    "foo" とタイプして、 "GoToFileMemberExample.Foo" に移動しましょう。
    //
    // 3. コンストラクターに移動します。
    //    "new" 、または "ctor" とタイプします。
    //
    // 4. 可視属性 (Visibility - public / private) で移動します。
    //    "public" 、または "private" とタイプします。


    // ----------

    public class GoToFileMemberExample
    {
        private readonly int bar;

        public GoToFileMemberExample(string foo, int bar)
        {
            this.bar = bar;
            Foo = foo;
        }

        public string Foo { get; private set; }
    }
}
